src/api/system.c

```diff
@@ -760,30 +745,46 @@ static int f_absolute_path(lua_State *L) {
 static int f_get_file_info(lua_State *L) {
   const char *path = luaL_checkstring(L, 1);
 
+  lua_newtable(L);
 #ifdef _WIN32
-  struct _stat s;
   LPWSTR wpath = utfconv_utf8towc(path);
   if (wpath == NULL) {
-    lua_pushnil(L);
-    lua_pushstring(L, UTFCONV_ERROR_INVALID_CONVERSION);
+    lua_pushnil(L); lua_pushstring(L, UTFCONV_ERROR_INVALID_CONVERSION);
+    return 2;
+  }
+  WIN32_FILE_ATTRIBUTE_DATA data;
+  if (!GetFileAttributesExW(wpath, GetFileExInfoStandard, &data)) {
+    free(wpath);
+    lua_pushnil(L); push_win32_error(L, GetLastError());
     return 2;
   }
-  int err = _wstat(wpath, &s);
   free(wpath);
+  ULARGE_INTEGER large_int = {0};
+  #define TICKS_PER_MILISECOND 10000
+  #define EPOCH_DIFFERENCE 11644473600000LL
+  // https://stackoverflow.com/questions/6161776/convert-windows-filetime-to-second-in-unix-linux
+  large_int.HighPart = data.ftLastWriteTime.dwHighDateTime; large_int.LowPart = data.ftLastWriteTime.dwLowDateTime;
+  lua_pushnumber(L, (double)((large_int.QuadPart / TICKS_PER_MILISECOND - EPOCH_DIFFERENCE)/1000.0));
+  lua_setfield(L, -2, "modified");
+
+  large_int.HighPart = data.nFileSizeHigh; large_int.LowPart = data.nFileSizeLow;
+  lua_pushinteger(L, large_int.QuadPart);
+  lua_setfield(L, -2, "size");
+
+  lua_pushstring(L, data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY ? "dir" : "file");
+  lua_setfield(L, -2, "type");
+
+  lua_pushboolean(L, data.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT);
+  lua_setfield(L, -2, "symlink");
```

Comment:
Does this get applied to files too?
Should we just set `symlink` to files on non-windows?

Reply:
What do you mean by this?